ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -580,6 +577,12 @@ private void processReplicaMetadataResponse(Set<StoreKey> missingRemoteStoreKeys
             notification.onBlobReplicaDeleted(dataNodeId.getHostname(), dataNodeId.getPort(), localKey.getID(),
                 BlobReplicaSourceType.REPAIRED);
           }
+        } else if (!deletedLocally && !messageInfo.isDeleted() && messageInfo.isTtlUpdated()) {
```

Comment:
Can `messageInfo.isDeleted()` and `messageInfo.isTtlUpdated()` both be true? Couldn't this check be `... else if (messageInfo.isTtlUpdated() && !deletedLocally) {}`?

Reply:
Yes it can be both ttl updated and deleted (just like an `IndexValue`) - the flags are additive. I think it is useful to store the history this way.